src/wallet/wallet2.cpp

```diff
@@ -559,6 +555,81 @@ void wallet2::set_seed_language(const std::string &language)
 {
   seed_language = language;
 }
+//----------------------------------------------------------------------------------------------------
+cryptonote::account_public_address wallet2::get_subaddress(size_t index) const
+{
+  if (index >= m_subaddresses_size)
+    throw std::runtime_error("requested index exceeds the limit! expand the hashtable by 'set subaddresses-size N'");
+  const cryptonote::account_keys& keys = m_account.get_keys();
+  if (index == 0)
+    return keys.m_account_address;
+  const crypto::secret_key& a = keys.m_view_secret_key;
+  const crypto::public_key& B = keys.m_account_address.m_spend_public_key;
+  uint64_t index64 = static_cast<uint64_t>(index);
+  char data[HASH_SIZE + 8];
+  memcpy(data            , &a      , HASH_SIZE);
+  memcpy(data + HASH_SIZE, &index64, 8);
+  crypto::secret_key m;
+  crypto::hash_to_scalar(data, HASH_SIZE + 8, m);
+  crypto::public_key M;
+  crypto::secret_key_to_public_key(m, M);
+  crypto::public_key D;
+  crypto::add_public_key(B, M, D);
+  crypto::public_key C;
+  crypto::secret_key_mult_public_key(a, D, C);
+  cryptonote::account_public_address address;
+  address.m_view_public_key  = C;
+  address.m_spend_public_key = D;
+  return address;
+}
+//----------------------------------------------------------------------------------------------------
+boost::optional<size_t> wallet2::find_subaddress(const crypto::public_key& spendkey) const
+{
+  auto found = m_subaddresses.find(spendkey);
+  if (found == m_subaddresses.end())
+    return boost::none;
+  return found->second;
+}
+//----------------------------------------------------------------------------------------------------
+std::pair<cryptonote::account_public_address, crypto::hash8> wallet2::make_disposable_address(size_t index) const
+{
+  if (index >= m_subaddresses_size)
+    throw std::runtime_error("requested index exceeds the limit! expand the hashtable by 'set subaddresses-size N'");
+  const crypto::secret_key a = m_account.get_keys().m_view_secret_key;
+  for (int i = 0; i < 100000; ++i)
```

Comment:
After reading the description for disposable addresses my first thought was how to generate one (because it seemed like brute force was necessary). And here it is. How often does this fail? How long does it take on average? Or even better, has anyone calculated the probability of finding such a value?